File: checky-py/checky.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
"""Lists how many images are in each brand's image processing folder"""
__author__ = "Jason Rebuck"
__copyright__ = "2020-2021"
__version__ = "0.16"

import os

def get_base_path():
    """Try to find proof's path"""
    for directory in ("/Volumes/proofs", "/mnt/proofs"):
        if os.path.isdir(directory):
            return directory
    kill_if_no_path() #pylint workaround
    return ""

def kill_if_no_path():
    """Warn and exit if no proofs path found."""
    print("\033[1m\033[31mMissing Proofs Path!\033[0m\n")
    exit()

def count(path):
    """Count how many files are in folder"""
    try:
        if os.path.isdir(path):
            return len([j for j in os.listdir(path) if j[0] != '.'])
        else:
            return -1
    except OSError:
        return 0

def print_title(brand):
    """Print brand title"""
    print(f"\n\033[7m{brand.upper():^20}\33[0m")

def print_count(name, num):
    """Print folder and num"""
    color = "\033[30m"
    icon = " "
    if num == 0:
        num = "▪"
    elif num == -1:
        num = "-ERR-"
        color = "\033[30m\033[2m"
    elif num >= 50:
        color = "\033[5m\033[35m" #blinky magenta
        icon = "☠️"
    elif num >= 40:
        color = "\033[32m" #green
        icon = "🐲"
    elif num >= 30:
        color = "\033[35m" #magenta
        icon = "🦄"
    elif num >= 20:
        color = "\033[33m" #yellow
        icon = "🔥"
    else:
        color = "\033[31m" #red
        icon = "📌"
    print(f" \033[1m{color}{name.upper():<14}{num:<2}\033[0m {icon}")

def print_status(brand, base, folders):
    """Loop through folders and run print count"""
    print_title(brand)
    for folder in folders:
        print_count(folder, count(f"{base}/images/bdp/{brand.lower()}/image_processing/{folder}"))

def main():
    """Print title and count of each brand and brand folder"""

    # Check if folder was mounted
    base_path = get_base_path()

    # HBD
    print_status("hbd", base_path, (
        "overlay", "portrait", "logo", "house",
        "pdf", "recipe", "postcard"))

    # YHL
    print_status("yhl", base_path, (
        "overlay", "portrait", "logo", "house",
        "pdf", "recipe", "postcard"))

    #RLP
    print_status("rlp", base_path, ("overlay", "portrait", "logo", "house", "pdf"))

    print("")

if __name__ == "__main__":
    main()

File: checky-go/checky.go
// """Lists how many images are in each brand's image processing folder"""
// __author__ = "Jason Rebuck"
// __copyright__ = "2022"
// __version__ = "0.05"

package main

import (
  "fmt"
  "os"
  "strings"
  . "github.com/logrusorgru/aurora"
)

func get_base_path() string {
  for _, dir := range [2]string{"/Volumes/proofs", "/mnt/proofs"} {
    if _, err := os.Stat(dir); !os.IsNotExist(err) {
      return dir
    }
  }
  kill_if_no_path()
  return ""
}

func kill_if_no_path(){
  fmt.Println(Red("Missing Proofs Path!"))
  fmt.Println("")
  os.Exit(0)
}

func count(path string) int {
  files, err := os.ReadDir(path)     
  if err != nil {
    panic(err)
  }
  return len(files)
}

func print_title(brand string) {
  fmt.Println(Inverse(fmt.Sprintf("%11s%-9s", strings.ToUpper(brand), "")))
}

func print_count(name string, num int) {
  color := Bold(strings.ToUpper(name))
  switch {
  case num == 0:
    fmt.Printf(" %-13s ▪ \n", color)
  case num == -1:
    fmt.Printf(" %-14s%-2d -ERR-\n", Faint(color), num)
  case num >=50:
    fmt.Printf(" %-14s%-2d ☠️ \n", Blink(Magenta(color)))
  case num >=40:
    fmt.Printf(" %-14s%-2d 🐲 \n", Green(color))
  case num >=30:
    fmt.Printf(" %-14s%-2d 🦄 \n", Magenta(color))
  case num >=20:
    fmt.Printf(" %-14s%-2d 🔥 \n", Yellow(color))
  default:
    fmt.Printf(" %-14s%-2d 📌 \n", Red(color))
  }
}

func print_status(brand string, base string, folders []string) {
  fmt.Println("")
  print_title(brand)
  for _, f := range folders {
    print_count(f, count(fmt.Sprintf("%s/images/bdp/%s/image_processing/%s", base, brand, f)))
  }
}

func main() {
  base_path := get_base_path()
  print_status("hbd", base_path, []string{"overlay", "portrait", "logo", "house", "pdf", "recipe", "postcard"})
  print_status("yhl", base_path, []string{"overlay", "portrait", "logo", "house", "pdf", "recipe", "postcard"})
  print_status("rlp", base_path, []string{"overlay", "portrait", "logo", "house", "pdf"})
  fmt.Println("")
}
